app/controllers/orders/pending.js

```diff
@@ -39,14 +39,9 @@ export default Controller.extend({
 
   actions: {
     aliPayCheckout(order_identifier) {
-      this.loader.post(`/create_source/${order_identifier}`)
-        .then(charge => {
-          if (charge.status) {
-            this.notify.success('Payment has succeeded');
-            this.transitionToRoute('orders.view', order_identifier);
-          } else {
-            this.notify.error('Payment has failed');
-          }
+      return this.loader.load(`alipay/create_source/${order_identifier}`)
```

Comment:
How are you handling a failure case ? When the network request fails for example or the server returns an error code.

Reply:
The failure case is handled on the server side. When the payment fails, a response is returned which specifies that the source creation/charge object has failed. https://github.com/fossasia/open-event-server/blob/development/app/api/orders.py#L500
After the payment gateways are merged, will be making the gateways agnostic so that they work properly with android as well.

Reply:
Not that failure case, lets say the server is unavailable. Or the server is throwing a 503. How are you handling that and showing an error to the user ? Here you are assuming only success (200) responses will be coming in.

Reply:
> Not that failure case, lets say the server is unavailable. Or the server is throwing a 503. How are you handling that and showing an error to the user ? Here you are assuming only success (200) responses will be coming in.
Oh yeah. That hasn't been handled here. I'll take care of that in the following iteration of this.

Reply:
Also, use async-await